homeassistant/components/sensibo/climate.py

```diff
@@ -242,27 +259,37 @@ def target_temperature_step(self) -> float | None:
     @property
     def fan_mode(self) -> str | None:
         """Return the fan setting."""
-        fan_mode: str | None = self.device_data.fan_mode
-        return fan_mode
+        if (fan_mode := self.device_data.fan_mode) in AVAILABLE_FAN_MODES:
+            return fan_mode
+        return None
 
     @property
     def fan_modes(self) -> list[str] | None:
         """Return the list of available fan modes."""
-        if self.device_data.fan_modes:
-            return self.device_data.fan_modes
+        if fan_modes := self.device_data.fan_modes:
+            fan_modes_set = set(fan_modes)
+            available_fan_modes = sorted(
+                set(fan_modes_set).intersection(AVAILABLE_FAN_MODES)
+            )
+            return list(available_fan_modes)
         return None
 
     @property
     def swing_mode(self) -> str | None:
         """Return the swing setting."""
-        swing_mode: str | None = self.device_data.swing_mode
-        return swing_mode
+        if (swing_mode := self.device_data.swing_mode) in AVAILABLE_SWING_MODES:
```

Comment:
Why not return it if it’s not an available mode ? Shouldn’t that be a warning if it is.

Reply:
For both `fan_mode` and `swing_mode` it now generates a warning to open an issue.
Still doesn't return the value if it's not with the available modes as I suppose it's not meant to (we want them to open an issue so we can add it, also properly translated)